File: app/src/main/java/com/gne/www/pinviewlibrary/MainActivity.java
package com.gne.www.pinviewlibrary;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.inputmethod.InputMethodManager;
import android.widget.Toast;

import com.gne.www.lib.InputType;
import com.gne.www.lib.OnPinCompletedListener;
import com.gne.www.lib.PinView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final PinView pinView=findViewById(R.id.pinview);

        pinView.setPinCount(6);
        pinView.setText("helllooo");
//        pinView.setPinBackground(getResources().getDrawable(R.drawable.pin_background));
        pinView.getText();
        pinView.setPinSize(50);
        pinView.setInputType(InputType.TYPE_NUMBER);
        pinView.setShowPasswordToggle(true);
        pinView.setPassword(true);
        pinView.requestPinFocus();
        pinView.requestPinFocus(4);
        pinView.setPasswordToggleColor(getResources().getColor(R.color.colorPrimary));
//        pinView.setPinSize(20);
        pinView.setPinTextSize(13);
        pinView.setPasswordToggleSize(20);


        pinView.setOnPinCompletionListener(new OnPinCompletedListener() {
            @Override
            public void onPinCompleted(String entirePin) {
                Toast.makeText(MainActivity.this,"Text Entered",Toast.LENGTH_LONG).show();
                hideKeyboard();
//                pinView.setShowPasswordToggle(false);
//                pinView.setInputType(InputType.TYPE_NUMBER);
            }
        });

    }

    private void hideKeyboard() {
        if (getCurrentFocus() != null) {
            InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
            inputMethodManager.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
        }
    }
}

File: app/src/main/java/com/darshan_miskin/www/pinviewlibrary/MainActivity.kt
package com.darshan_miskin.www.pinviewlibrary

import android.os.Bundle
import android.util.Log
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.darshan_miskin.www.lib.InputType
import com.darshan_miskin.www.lib.PinView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val pinView = findViewById<PinView>(R.id.pinview)

        pinView.pinCount = 4
        pinView.text = "hello!!"
        pinView.pinBackground = ContextCompat.getDrawable(this,R.drawable.pin_background)
        pinView.isPassword = false
        pinView.text
        pinView.pinSizeDp = 40
        pinView.inputType = InputType.TEXT
        pinView.showPasswordToggle = true
        pinView.requestPinFocusAt(4)
        pinView.passwordToggleColor = ContextCompat.getColor(this, R.color.colorPrimary)
        pinView.pinTextSizeSp = 13f
        pinView.passwordToggleSizeDp = 30
        pinView.pinTextColor = ContextCompat.getColor(this, R.color.colorPrimary)

        pinView.setOnPinChangedListener {
            Log.d("asdf", "Pin Changed to: ${pinView.text}")
        }

        pinView.setOnPinCompletionListener { entirePin ->
            Toast.makeText(this@MainActivity, "Entire pin is: $entirePin", Toast.LENGTH_LONG).show()
            hideKeyboard()
        }
    }

    private fun hideKeyboard() {
        if (currentFocus != null) {
            val inputMethodManager = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
            inputMethodManager.hideSoftInputFromWindow(currentFocus?.windowToken, 0)
        }
    }
}
